# Codveda Technologies Internship

# Level 1 - Task 1: Data Cleaning and Preprocessing

### Intern Name:
**Ashvini Manohar Jadhav**

### Domain:
**Data Analysis**

### Tools Used:
- Python
- Pandas
- NumPy
- Jupyter Notebook

### Dataset:
House Prediction Dataset (Boston Housing Dataset)

## Project Objective

The objective of this task is to clean and preprocess the House Prediction dataset before performing data analysis.

The preprocessing steps include:

- Loading the dataset
  
- Understanding dataset structure

  
- Checking data types

  
- Identifying missing values

  
- Detecting duplicate records

  
- Verifying data consistency

  
- Saving the cleaned dataset

In [1]:
import pandas as pd
import numpy as np

## Step 1: Load the Dataset

In [27]:

column_names = [
    'CRIM', 'ZN', 'INDUS', 'CHAS', 'NOX', 'RM', 'AGE',
    'DIS', 'RAD', 'TAX', 'PTRATIO', 'B', 'LSTAT', 'MEDV'
]

df = pd.read_csv("house_prediction.csv" ,sep=r"\s+", header=None ,names=column_names)

df.head()

,CRIM,ZN,INDUS,CHAS,NOX,RM,AGE,DIS,RAD,TAX,PTRATIO,B,LSTAT,MEDV
0,0.00632,18.0,2.31,0,0.538,6.575,65.2,4.0900,1,296.0,15.3,396.90,4.98,24.0
1,0.02731,0.0,7.07,0,0.469,6.421,78.9,4.9671,2,242.0,17.8,396.90,9.14,21.6
2,0.02729,0.0,7.07,0,0.469,7.185,61.1,4.9671,2,242.0,17.8,392.83,4.03,34.7
3,0.03237,0.0,2.18,0,0.458,6.998,45.8,6.0622,3,222.0,18.7,394.63,2.94,33.4
4,0.06905,0.0,2.18,0,0.458,7.147,54.2,6.0622,3,222.0,18.7,396.90,5.33,36.2


## Step 2: Dataset Overview

In [18]:
# Display the shape of the dataset
print("Shape of Dataset:", df.shape)

Shape of Dataset: (506, 14)


In [19]:
# Display column names
print(df.columns)

Index(['CRIM', 'ZN', 'INDUS', 'CHAS', 'NOX', 'RM', 'AGE', 'DIS', 'RAD', 'TAX',
       'PTRATIO', 'B', 'LSTAT', 'MEDV'],
      dtype='str')


In [20]:
# Display dataset information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 14 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   CRIM     506 non-null    float64
 1   ZN       506 non-null    float64
 2   INDUS    506 non-null    float64
 3   CHAS     506 non-null    int64  
 4   NOX      506 non-null    float64
 5   RM       506 non-null    float64
 6   AGE      506 non-null    float64
 7   DIS      506 non-null    float64
 8   RAD      506 non-null    int64  
 9   TAX      506 non-null    float64
 10  PTRATIO  506 non-null    float64
 11  B        506 non-null    float64
 12  LSTAT    506 non-null    float64
 13  MEDV     506 non-null    float64
dtypes: float64(12), int64(2)
memory usage: 55.5 KB


### Observation

- The dataset contains **506 rows** and **14 columns**.
  
- All columns were loaded successfully.

  
- Data types are numerical.

## Step 3: Summary Statistics

In [21]:
# Statistical summary
df.describe()

,CRIM,ZN,INDUS,CHAS,NOX,RM,AGE,DIS,RAD,TAX,PTRATIO,B,LSTAT,MEDV
count,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000,506.000000
mean,3.613524,11.363636,11.136779,0.069170,0.554695,6.284634,68.574901,3.795043,9.549407,408.237154,18.455534,356.674032,12.653063,22.532806
std,8.601545,23.322453,6.860353,0.253994,0.115878,0.702617,28.148861,2.105710,8.707259,168.537116,2.164946,91.294864,7.141062,9.197104
min,0.006320,0.000000,0.460000,0.000000,0.385000,3.561000,2.900000,1.129600,1.000000,187.000000,12.600000,0.320000,1.730000,5.000000
25%,0.082045,0.000000,5.190000,0.000000,0.449000,5.885500,45.025000,2.100175,4.000000,279.000000,17.400000,375.377500,6.950000,17.025000
50%,0.256510,0.000000,9.690000,0.000000,0.538000,6.208500,77.500000,3.207450,5.000000,330.000000,19.050000,391.440000,11.360000,21.200000
75%,3.677083,12.500000,18.100000,0.000000,0.624000,6.623500,94.075000,5.188425,24.000000,666.000000,20.200000,396.225000,16.955000,25.000000
max,88.976200,100.000000,27.740000,1.000000,0.871000,8.780000,100.000000,12.126500,24.000000,711.000000,22.000000,396.900000,37.970000,50.000000


## Step 4: Check Missing Values

In [28]:
# Check missing values
missing_values = df.isnull().sum()

print(missing_values)

CRIM       0
ZN         0
INDUS      0
CHAS       0
NOX        0
RM         0
AGE        0
DIS        0
RAD        0
TAX        0
PTRATIO    0
B          0
LSTAT      0
MEDV       0
dtype: int64


### Observation

No missing values were found in the dataset.

Therefore, no imputation or deletion of records was required.

## Step 5: Check Duplicate Records

In [29]:
duplicates = df.duplicated().sum()

print("Duplicate Rows:", duplicates)

Duplicate Rows: 0


### Observation

No duplicate records were found in the dataset.

Therefore, no duplicate rows were removed.

## Step 6: Verify Data Types

In [24]:
# Display data types
df.dtypes

CRIM       float64
ZN         float64
INDUS      float64
CHAS         int64
NOX        float64
RM         float64
AGE        float64
DIS        float64
RAD          int64
TAX        float64
PTRATIO    float64
B          float64
LSTAT      float64
MEDV       float64
dtype: object

In [25]:
# Check unique values in CHAS
print("Unique values in CHAS:", df["CHAS"].unique())

# Check data types again
print(df.dtypes)

Unique values in CHAS: [0 1]
CRIM       float64
ZN         float64
INDUS      float64
CHAS         int64
NOX        float64
RM         float64
AGE        float64
DIS        float64
RAD          int64
TAX        float64
PTRATIO    float64
B          float64
LSTAT      float64
MEDV       float64
dtype: object


### Observation

All columns have appropriate numerical data types.

No data type conversion was required.

## Step 7: Save the Cleaned Dataset

In [26]:
# Save cleaned dataset
df.to_csv("cleaned_house_prediction.csv", index=False)

print("Cleaned dataset saved successfully!")

Cleaned dataset saved successfully!


# Conclusion

The House Prediction dataset was successfully loaded and inspected.

### Completed preprocessing tasks:

- Loaded the dataset using Pandas.
    
- Verified the dataset structure.
    
- Checked data types.
    
- Checked for missing values.
    
- Checked for duplicate records.
    
- Verified data consistency.
    
- Saved the cleaned dataset.

The dataset was already clean and ready for further analysis.